apps/prairielearn/src/pages/instructorAssessmentManualGrading/assessmentQuestion/assessmentQuestionTable.tsx

```diff
@@ -0,0 +1,781 @@
+        return;
+      }
+      const input = form.querySelector<HTMLInputElement>('input:not([type="hidden"])');
+      if (input) {
+        input.focus();
+      }
+
+      // Remove any existing event listeners to prevent duplicates
+      form.removeEventListener('submit', handlePointsFormSubmit);
+      form.addEventListener('submit', handlePointsFormSubmit);
+    }
+
+    // Attach event listeners to all popover trigger buttons
+    const popoverButtons = document.querySelectorAll('[data-bs-toggle="popover"]');
+    popoverButtons.forEach((button) => {
+      button.addEventListener('shown.bs.popover', handlePopoverShown);
+    });
+
+    // Cleanup function
+    return () => {
+      const form = document.querySelector<HTMLFormElement>('form[name=edit-points-form]');
+      form?.removeEventListener('submit', handlePointsFormSubmit);
+
+      popoverButtons.forEach((button) => {
+        button.removeEventListener('shown.bs.popover', handlePopoverShown);
+      });
+    };
+  }, [queryClientInstance, urlPrefix, assessmentId, assessmentQuestion.id, instanceQuestions]);
+
+  // Use batch actions hook
+  const {
+    batchActionMutation,
+    handleBatchAction,
+    deleteAiGradingJobsMutation,
+    deleteAiGroupingsMutation,
+  } = useBatchActions({
+    csrfToken,
+    urlPrefix,
+    assessmentId,
+    assessmentQuestionId: assessmentQuestion.id,
+    setErrorMessage,
+    setSuccessMessage,
+  });
+
+  const columnFiltersComponents = createColumnFilters({
+    allGraders,
+    allSubmissionGroups,
+    allAiAgreementItems,
+    aiGradingMode,
+    instanceQuestionGroups,
+    assessmentQuestion,
+    gradingStatusFilter,
+    setGradingStatusFilter,
+    assignedGraderFilter,
+    setAssignedGraderFilter,
+    gradedByFilter,
+    setGradedByFilter,
+    submissionGroupFilter,
+    setSubmissionGroupFilter,
+    aiAgreementFilter,
+    setAiAgreementFilter,
+    manualPointsFilter,
+    setManualPointsFilter,
+    autoPointsFilter,
+    setAutoPointsFilter,
+    totalPointsFilter,
+    setTotalPointsFilter,
+    scoreFilter,
+    setScoreFilter,
+  });
+
+  return (
+    <>
+      <div class="mb-3">
+        <RubricSettings
+          assessmentQuestion={assessmentQuestion}
+          rubricData={rubricData}
+          csrfToken={csrfToken}
+          aiGradingStats={aiGradingStats}
+          context={{
+            course_short_name: course.short_name,
+            course_instance_short_name: courseInstance.short_name,
+            assessment_tid: assessmentTid,
+            question_qid: questionQid,
+          }}
+        />
+      </div>
+      {errorMessage && (
+        <Alert variant="danger" class="mb-3" dismissible onClose={() => setErrorMessage(null)}>
+          <strong>Error:</strong> {errorMessage}
+        </Alert>
+      )}
+      {successMessage && (
+        <Alert variant="success" class="mb-3" dismissible onClose={() => setSuccessMessage(null)}>
+          {successMessage}
+        </Alert>
+      )}
```

Comment:
This structure of state allows us to have a success message and an error message at the same time? Do you consider that to be a valid state? It doesn't seem like it for me, so consider [making invalid states unrepresentable](https://geeklaunch.io/blog/make-invalid-states-unrepresentable/) and using something like this instead:

```ts
type AlertMessage = { type: 'success' | 'error', message: string };
const [alertMessage, setAlertMessage] = useState<AlertMessage | null>(null);
```